.github/workflows/codeql.yaml

```diff
@@ -0,0 +1,43 @@
+---
+name: codeQL
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Static analysis with CodeQL
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
```

Comment:
I don't see the need of adding a timeout this low, may break the workflow for wrong reasons
```suggestion
```